controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -386,6 +382,22 @@ public ActivateResult deploy(ApplicationId applicationId, ZoneId zone,
         }
     }
 
+    /** Fetches the requested application package from the artifact store(s). */
+    public ApplicationPackage getApplicationPackage(Application application, ApplicationVersion version) {
+        try {
+            return application.deploymentJobs().deployedInternally()
+                    ? new ApplicationPackage(applicationStore.get(application.id(), version))
```

Comment:
Consider creating two locally scoped suppliers (`Supplier<ApplicationPackage>`), one for each store, and then use those here and below.

Reply:
This kludge will be gone soon, and I'll clean up then :)